pyccel/stdlib/ndarrays/ndarrays.c

```diff
@@ -373,3 +425,71 @@ int64_t     *numpy_to_ndarray_shape(int64_t *np_shape, int nd)
     return nd_shape;
 
 }
+
+void array_copy_data(t_ndarray dest, t_ndarray src, int64_t pad)
+{
+    int64_t index = 0;
+    int64_t shape_product_src = 1;
+    int64_t shape_product_dest = 1;
+    int64_t var = 0;
+    while (index < dest.length)
+    {
+        int64_t ndim_src = src.nd - 1;
+        int64_t ndim_dest = dest.nd - 1;
+        int64_t i_src = (index % src.shape[ndim_src]) * src.strides[ndim_src];
+        int64_t i_dest = (index % dest.shape[ndim_dest]) * dest.strides[ndim_dest];
+        ndim_src--;
+        ndim_dest--;
+        while (ndim_src >= 0)
+        {
+            // i_src calculation
+            shape_product_src *= src.shape[ndim_src + 1];
+            var = index / shape_product_src;
+            if(var >= src.shape[ndim_src])
+                var = var % src.shape[ndim_src];
+            i_src += (var) * src.strides[ndim_src];
+            ndim_src--;
+            // i_dest calculation
+            shape_product_dest *= dest.shape[ndim_dest + 1];
+            var = index / shape_product_dest;
+            if(var >= dest.shape[ndim_dest])
+                var = var % dest.shape[ndim_dest];
+            i_dest += (var) * dest.strides[ndim_dest];
+            ndim_dest--;
+
+        }
+        switch (dest.type)
+        {
+            case nd_int8:
+                dest.nd_int8[i_dest + pad] = src.nd_int8[i_src];
+                break;
+            case nd_int16:
+                dest.nd_int16[i_dest + pad] = src.nd_int16[i_src];
+                break;
+            case nd_int32:
+                dest.nd_int32[i_dest + pad] = src.nd_int32[i_src];
+                break;
+            case nd_int64:
+                dest.nd_int64[i_dest + pad] = src.nd_int64[i_src];
+                break;
+            case nd_float:
+                dest.nd_float[i_dest + pad] = src.nd_float[i_src];
+                break;
+            case nd_double:
+                dest.nd_double[i_dest + pad] = src.nd_double[i_src];
+                break;
+            case nd_bool:
+                dest.nd_bool[i_dest + pad] = src.nd_bool[i_src];
+                break;
+            case nd_cfloat:
+                dest.nd_cfloat[i_dest + pad] = src.nd_cfloat[i_src];
+                break;
+            case nd_cdouble:
+                dest.nd_int64[i_dest + pad] = src.nd_cdouble[i_src];

```

Comment:
Maybe `pad` should be called `offset`? Padding usually implies the insertion of a certain value (often spaces) around the data